node/pkg/lifecycle/startup/startup.go

```diff
@@ -182,25 +182,22 @@ func Run() {
 		}
 	}
 
-	// If Calico is running in policy only mode we don't need to write BGP related details to the Node.
-	if os.Getenv("CALICO_NETWORKING_BACKEND") != "none" {
-		configureAndCheckIPAddressSubnets(ctx, cli, node, k8sNode)
-		// Configure the node AS number.
-		configureASNumber(node)
-	}
-
+	needsNodeUpdate := configureAndCheckIPAddressSubnets(ctx, cli, node, k8sNode)
+	// Configure the node AS number.
+	needsNodeUpdate = needsNodeUpdate || configureASNumber(node)
```

Comment:
I think this line will skip execution of `configureASNumber` if `needsNodeUpdate` is true.

We probably want to do it like this instead.

```go
needsNodeUpdate = configureASNumber(node) || needsNodeUpdate
```

So that `configureASNumber` is always called

Reply:
Thanks for catching this, revered the order.